### PR TITLE
client_manage(): Check that reparenting works

### PR DESCRIPTION
**Pull request by psychon**
Testing done: Basically none (`tests/run.sh` did not complain)

------

When a new client appears, we reparent its window into a window of our
choice so that we can add decorations. However, windows can be difficult
making the reparenting fail. For example, some snapshots of GTK+ set a
parent-relative background, which means that its parent windows have to
have the same depth as the window itself. This makes reparenting fail if
transparency is used/supported.

Make the failure mode... different, by checking if reparenting
succeeded. If it failed, we print an error message and unmanage the
window immediately. This means that those GTK+ windows just do not
appear at all, instead of behaving weirdly.

This idea of this patch came from i3.

Related-to: https://github.com/awesomeWM/awesome/issues/2279
Signed-off-by: Uli Schlachter <psychon@znc.in>

### CODE REVIEW

File: objects/client.c [hunk 1]
```diff
@@ -1533,6 +1534,14 @@ client_manage(xcb_window_t w, xcb_get_geometry_reply_t *wgeom, xcb_get_window_at
     luaA_object_emit_signal(L, -1, "manage", 0);
     /* pop client */
     lua_pop(L, 1);
+
+    xcb_generic_error_t *error = xcb_request_check(globalconf.connection, reparent_cookie);
+    if (error != NULL) {
+        warn("Failed to manage some window, because reparenting failed.");
```
Comment (blueyed) [inline objects/client.c]: Could we be more specific here instead of "some window"?

Comment (psychon) [reply objects/client.c]: The window id will not help anybody. Should `c->name` be included in the message?

I think that when this triggers, it is obvious that something is wrong (you start firefox, no window appears, but a firefox process is running). The harder part would be for people to actually find this warning. But yes, that's no excuse to produce a bad warning.

Comment (blueyed) [reply objects/client.c]: Ok with me then.

**Comment by psychon** [comment]
> psychon dismissed stale reviews from blueyed and actionless via f3e62b1 just now

Diff to previous version:
```diff
diff --git a/objects/client.c b/objects/client.c
index a4d3c42ab..e1c82e2de 100644
--- a/objects/client.c
+++ b/objects/client.c
@@ -1532,16 +1532,18 @@ client_manage(xcb_window_t w, xcb_get_geometry_reply_t *wgeom, xcb_get_window_at
 
     /* client is still on top of the stack; emit signal */
     luaA_object_emit_signal(L, -1, "manage", 0);
-    /* pop client */
-    lua_pop(L, 1);
 
     xcb_generic_error_t *error = xcb_request_check(globalconf.connection, reparent_cookie);
     if (error != NULL) {
-        warn("Failed to manage some window, because reparenting failed.");
+        warn("Failed to manage window with name '%s', class '%s', instance '%s', because reparenting failed.",
+                NONULL(c->name), NONULL(c->class), NONULL(c->instance));
         event_handle((xcb_generic_event_t *) error);
         p_delete(&error);
         client_unmanage(c, true);
     }
+
+    /* pop client */
+    lua_pop(L, 1);
 }
 
 static void
```

File: objects/client.c (hunk 1 above)
Comment (Elv13) [reply objects/client.c]: Ah, right, I forgot about those

**Pull request merged**